### DISCUSSION TITLE
Two rules with same starting token, but different ending - throws conflict error

### DISCUSSION THREAD

**Discussion by redexp** [tree-sitter/tree-sitter#3401] [Q&A] [answered]
Good day

I have syntax like
```
name

name = name
```
* `name` by it's own `group_start: $ => seq($.name, $._new_line_or_eof)` is a start and name of group. 
* `name = ...` - `group_items: $ => seq($.name, '=', $.name, $._new_line_or_eof)` are items of this group
* group described as `group: $ => seq($.group_start, repeat($.group_items))`
* whole file `document: $ => repeat($.group)`

tree-sitter find out conflict with `name _new_line_or_eof name ...` and I can't make him to look ahead if there `_new_line_or_eof` then it's new group, if `=` then it's `group_item`

I can fix it with: 
* adding new syntax like divider `-----` before each group, but I really don't want it
* include `\n` to `group_start` regex like `group_start: _ => /\w+\n/` so it's different with `name: _ => /\w+/`, but I want `group_start` to have clean name without `\n`. Also it will be error if file ends with `name` but without `\n`

I'v tried to search for solution in this discussions, tried to find something similar in `TOML` and `JavaScript` grammars but everybody just adding keywords or brackets

**Comment by RedCMD** [comment]
use `prec()` around the 2nd item in both `seq()`'s

**Reply by redexp** [reply]
Thank you for response, but unfortunately it doesn't help. 
```
group_start: $ => seq($.name, prec(1, $._new_line_or_eof)),
group_items: $ => seq($.name, '=', $.name, prec(1, $._new_line_or_eof)),
```

**Reply by RedCMD** [reply]
how are you matching `EOF`?

```
1:  Specify a left or right associativity in `group`
2:  Add a conflict for these rules: `group`
```
try `prec.right()` in `group`
```js
document: $ => repeat($.group),
group: $ => prec.right(seq($.group_start, repeat($.group_items))),
group_start: $ => seq($.name, $._new_line_or_eof),
group_items: $ => seq($.name, '=', $.name, $._new_line_or_eof),
name: $ => /\w+/,
_new_line_or_eof: $ => /\r\n?/,
```

**Reply by redexp** [reply]
Thank you for helping me out

> how are you matching EOF?

I took it from [scanner.c of toml](https://github.com/ikatyang/tree-sitter-toml/blob/8bd2056818b21860e3d756b5a58c4f6e05fb744e/src/scanner.c#L66) 

so `_new_line_or_eof` is in `externals`
```
externals: $ => [$._new_line_or_eof],
```

> try prec.right() in group

I'v tried - it resolves conflict but parser can't understand group start now, it thinks that name by it's own is start of group_item and resulting tree has error

**Reply by redexp** [reply, marked as answer]
Just added `conflicts: $ => [[$.group]]` and everything fixes magically
